### PR TITLE
modulesync 3.0.0 & puppet-lint fixes & rubocop fixes

### PR DESCRIPTION
**Pull request by bastelfreak**

### CODE REVIEW

**Comment by dhollinger** [comment]
@bastelfreak Could you look into the failing tests?

**Comment by bastelfreak** [comment]
oh why doesn't centos apply anymore :(

**Comment by ekohl** [comment]
Was there a version change?

```
Notice: /Stage[main]/Zabbix::Database::Postgresql/Exec[zabbix_server_create.sql]/returns: /bin/sh: line 0: cd: /usr/share/doc/zabbix-*-pgsql-3.0*/: No such file or directory
```

File: spec/spec_helper_acceptance.rb [hunk 1]
```diff
@@ -1,11 +1,6 @@
+# This file is completely managed via modulesync
 require 'voxpupuli/acceptance/spec_helper_acceptance'
 
-configure_beaker do |host|
-  # The CentOS docker image has a yum config that won't install docs, to keep used space low
```
Comment (baurmatt) [inline spec/spec_helper_acceptance.rb:4]: Hey @bastelfreak, I think removing those lines might be the reason why the tests fails.

Comment (bastelfreak) [reply spec/spec_helper_acceptance.rb:4]: oh right, I can even remember that people debugged that on IRC. the docs packages contain the sql files.

File: spec/spec_helper_acceptance.rb [hunk 2]
```diff
@@ -1,11 +1,6 @@
+# This file is completely managed via modulesync
```
Comment (ekohl) [inline spec/spec_helper_acceptance.rb:1]: I'd keep it as it was. The default is really only 2 lines so IMHO there's no benefit in syncing it. Much cleaner to just do the node setup once than to do it in every single test.

**Review by ekohl — CHANGES_REQUESTED** [review]

File: spec/acceptance/zabbix_application_spec.rb [hunk 3]
```diff
@@ -1,6 +1,7 @@
 require 'spec_helper_acceptance'
 require 'serverspec_type_zabbixapi'
 
+# rubocop:disable RSpec/LetBeforeExamples
```
Comment (alexjfisher) [inline spec/acceptance/zabbix_application_spec.rb:4]: Did this need to be disabled?

Comment (bastelfreak) [reply spec/acceptance/zabbix_application_spec.rb:4]: I assume so. Rubocop complained that this line is now useless and needs to be removed.

Comment (bastelfreak) [reply spec/acceptance/zabbix_application_spec.rb:4]: ah, wrong comment. yes, rubocop said there are multiple issues for that cop. I wasn't able to fix this quickly and we really need to get this modulesync merged, so I disabled it.

**Review by dhollinger — APPROVED** [review]

File: spec/support/acceptance/prepare_host.rb [hunk 4]
```diff
@@ -1,5 +1,10 @@
-def cleanup_zabbix
-  shell('rm -rf /etc/yum.repos.d/Zabbix*.repo; rm -rf /var/cache/yum/x86_64/*/Zabbix*; yum clean all --verbose') if fact('os.family') == 'RedHat'
+def prepare_host
+  if fact('os.family') == 'RedHat'
+    shell('rm -rf /etc/yum.repos.d/Zabbix*.repo; rm -rf /var/cache/yum/x86_64/*/Zabbix*; yum clean all --verbose')
+    if fact('os.release.major').to_i == 7
+      shell('sed -i "/nodocs/d" /etc/yum.conf')
+    end
```
Comment (ekohl) [inline spec/support/acceptance/prepare_host.rb:6]: I really wanted the comment to survive.

Comment (bastelfreak) [reply spec/support/acceptance/prepare_host.rb:6]: oh right, my bad. fixed it in https://github.com/voxpupuli/puppet-zabbix/pull/710

**Pull request merged**